toolManager.lua

```diff
@@ -399,86 +399,164 @@ function courseplay:resetTipTrigger(vehicle, changeToForward)
 	end
 end;
 
+WorkWidthUtil = {}
+
 --- Iterator for all work areas of an object
-function courseplay:workAreaIterator(object)
+---@param object table
+function WorkWidthUtil.workAreaIterator(object)
 	local i = 0
 	return function()
 		i = i + 1
-		local wa = object and object.getWorkAreaByIndex and object:getWorkAreaByIndex(i)
+		local wa = WorkWidthUtil.getWorkAreas(object)
```

Comment:
This is not the same as far as I can tell, getWorkAreas only returns the first work area now.